solidity/contracts/bridge/WalletCoordinator.sol

```diff
@@ -0,0 +1,617 @@
+    ///      keccak256(fundingTxHash | fundingOutputIndex) for gas efficiency.
+    struct DepositKey {
+        bytes32 fundingTxHash;
+        uint32 fundingOutputIndex;
+    }
+
+    /// @notice Helper structure holding deposit extra data required during
+    ///         deposit sweep proposal validation. Basically, this structure
+    ///         is a combination of BitcoinTx.Info and relevant parts of
+    ///         Deposit.DepositRevealInfo.
+    /// @dev These data can be pulled from respective `DepositRevealed` events
+    ///      emitted by the `Bridge.revealDeposit` function. The `fundingTx`
+    ///      field must be taken directly from the Bitcoin chain, using the
+    ///      `DepositRevealed.fundingTxHash` as transaction identifier.
+    struct DepositExtraInfo {
+        BitcoinTx.Info fundingTx;
+        bytes8 blindingFactor;
+        bytes20 walletPubKeyHash;
+        bytes20 refundPubKeyHash;
+        bytes4 refundLocktime;
+    }
+
+    /// @notice Mapping that holds addresses allowed to submit proposals.
+    mapping(address => bool) public isProposalSubmitter;
+
+    /// @notice Mapping that holds wallet time locks. The key is a 20-byte
+    ///         wallet public key hash. The value is a UNIX timestamp defining
+    ///         the moment until which the wallet is locked and cannot receive
+    ///         new proposals. The value of 0 means the wallet is not locked
+    ///         and can receive a proposal at any time.
+    mapping(bytes20 => uint32) public walletLock;
+
+    /// @notice Handle to the Bridge contract.
+    Bridge public bridge;
+
+    /// @notice Handle to the WalletRegistry contract.
+    EcdsaWalletRegistry public walletRegistry;
+
+    /// @notice Determines the deposit sweep proposal validity time. In other
+    ///         words, this is the worst-case time for a deposit sweep during
+    ///         which the wallet is busy and cannot take another actions. This
+    ///         is also the duration of the time lock applied to the wallet
+    ///         once a new deposit sweep proposal is submitted.
+    ///
+    ///         For example, if a deposit sweep proposal was submitted at
+    ///         2 pm and depositSweepProposalValidity is 4 hours, the next
+    ///         proposal (of any type) can be submitted after 6 pm.
+    uint32 public depositSweepProposalValidity;
+
+    /// @notice The minimum time that must elapse since the deposit reveal
+    ///         before a deposit becomes eligible for a deposit sweep.
+    ///
+    ///         For example, if a deposit was revealed at 9 am and depositMinAge
+    ///         is 2 hours, the deposit is eligible for sweep after 11 am.
+    ///
+    /// @dev Forcing deposit minimum age ensures block finality for Ethereum.
+    ///      In the happy path case, i.e. where the deposit is revealed immediately
+    ///      after being broadcast on the Bitcoin network, the minimum age
+    ///      check also ensures block finality for Bitcoin.
```

Comment:
That's quite a good point for [this](https://github.com/keep-network/tbtc-v2/pull/575#discussion_r1151685038) discussion.  We can't trust `depositMinAge` for Bitcoin finality validation.  👌🏻